runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1531,7 +1534,47 @@ NNFW_STATUS nnfw_session::train_export_circle(const char *path)
   if (!mmapfile.ensure_mmap())
     return NNFW_STATUS_ERROR;
 
-  // TODO: Update weights in mmapfile
+  try
+  {
+    _execution->iterateTrainableTensors([&](const onert::ir::OperandIndex &idx,
+                                            const onert::backend::train::ITrainableTensor *tensor) {
+      auto model = ::circle::GetModel(mmapfile.buf());
+      auto subgs = model->subgraphs();
+      if (!model || !subgs || subgs->size() != 1)
+        throw std::runtime_error("Circle does not has valid subgraph or has multiple subgraphs");
+
+      auto subg = subgs->Get(0); // Get 1st subgraph
+      if (!idx.valid() || idx.value() >= subg->tensors()->size())
+        throw std::runtime_error("Trainable tensor index is out of range");
+
+      auto buf_idx = subg->tensors()->Get(idx.value())->buffer();
+      const ::circle::Buffer *buffer = (*model->buffers())[buf_idx];
+      if (!buffer || !buffer->data())
+        throw std::runtime_error("Buffer for trainable tensors is invalid");
+
+      const flatbuffers::Vector<uint8_t> *array = buffer->data();
+      if (!array)
+        throw std::runtime_error("Data for trainable tensor's buffer is invalid");
+
+      uint8_t *org_buf = const_cast<uint8_t *>(array->Data());
```

Comment:
I suggest to use const data type here and use `const_cast` in `memcpy` function directly so that the pointer address does not change in the middle.

Reply:
@jyoungyun I don't understand what you're concerning by _the pointer address does not change in the middle_. Also, as you can see a few lines below, I would like to check whether `org_buf` is null or not before passing it to `memcpy`.

Anyway, I applied my understanding of your suggestion.